plugins/inputs/snmp/snmp.go

```diff
@@ -535,8 +538,8 @@ func (t Table) Build(gs snmpConnection, walk bool) (*RTable, error) {
 		Time: time.Now(), //TODO record time at start
 		Rows: make([]RTableRow, 0, len(rows)),
 	}
-	for _, r := range rows {
-		if len(r.Tags) < tagCount {
+	for i, r := range rows {
+		if len(r.Tags)+emptyTags[i] < tagCount {
 			// don't add rows which are missing tags, as without tags you can't filter
 			continue
 		}
```

Comment:
Yes, like what I just implemented:

From a user's point of view, I would say `rtr4` is a broken entry in the table and I wouldn't want it to end up in the database. In contrast, an empty value might be a valid tag. In my case it was a description which I am totally fine with being empty as long as it exists.

But I can deal with either way as long as the tables I'm interested in are always sane...